mapbook-iOS/View controllers/Portal Browser/PortalBrowserViewController.swift

```diff
@@ -68,62 +65,98 @@ class PortalItemsListViewController: UIViewController {
      method. And simply refreshes the table view on successful completion
      or displays the error on failure.
     */
-    fileprivate func fetchPortalItems(using keyword:String?) {
+    fileprivate func fetchPortalItems(using keyword: String?) {
+        
+        if packageFinder == nil {
```

Comment:
This one was a tough call for me. One one hand the pattern you are outlining above is a safe swift pattern. On the other hand, i'm trying to establish a pattern where `packageManager` is not optional (in the true definition of it). Because of how Storyboards work, it isn't possible to provide `packageManager` at initialization. Therefore, I require that the app parent vc provides it as it prepares for the segue. What this `preconditionFailure` is attempting to communicate is that "this scenario should logically never be reached and if it is, something very important in the code has changed."

Can I introduce the guard statement and then leave as-is? I'd rather the app bomb-out in tests that to work around a required property by giving it the option to be optional.

Reply:
Ok, yes, that makes sense.  Having the guard and leaving it as-is is fine.